components/service/glean/src/main/java/mozilla/components/service/glean/BooleanMetricType.kt

```diff
@@ -58,8 +62,16 @@ data class BooleanMetricType(
      *                 value in [sendInPings]
      * @return true if metric value exists, otherwise false
      */
+    @Suppress("MagicNumber")
     @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     fun testHasValue(pingName: String = getStorageNames().first()): Boolean {
+        if (ioTask != null) {
+            var loopCnt = 50
+            while (ioTask!!.isActive && loopCnt > 0) {
+                --loopCnt
+            }
+        }
```

Comment:
Also, maybe this code block could be put in a helper function (where `ioTask` is passed in) to avoid the copy-and-paste.

Reply:
I like it!